ui/nuxeo-page-provider-display-behavior.js

```diff
@@ -587,10 +595,36 @@ export const PageProviderDisplayBehavior = [
         this.nxProvider.offset = 0;
         return this.nxProvider.fetch(options).then((response) => {
           if (page === 1) {
-            this.reset();
+            // TODO - maybe we can refactor this condition block
+            // get results count, and reset the array if it differs from current array length
+            let count;
+            if (response.resultsCount < 0) {
+              // negative resultCount means unknown value, fall back on currentPageSize
+              count = response.resultsCountLimit > 0 ? response.resultsCountLimit : response.currentPageSize;
+            } else if (response.resultsCountLimit > 0 && response.resultsCountLimit < response.resultsCount) {
+              count = response.resultsCountLimit;
+            } else {
+              count = response.resultsCount;
+            }
+            if (this.maxItems) {
+              if (count > this.maxItems) {
+                count = this.maxItems;
+              }
+            }
```

Comment:
```suggestion
            const { currentPageSize, resultsCount, resultsCountLimit } = response;
            if (resultsCount < 0) {
              // negative resultCount means unknown value, fallback on currentPageSize
              count = resultsCountLimit > 0 ? resultsCountLimit : currentPageSize;
            } else {
              count = (resultsCountLimit > 0 && resultsCountLimit < resultsCount) ? resultsCountLimit : resultsCount;
            }
            count = (this.maxItems && count > this.maxItems) ? this.maxItems : count;
```
Might not be worth it since it's not really refactored, just made shorter.

Reply:
Well, it can be one step forward with the refactoring. Thanks, I'll review it.